plugins/aws/appmesh/appmeshVGHealthChecks.js

```diff
@@ -0,0 +1,94 @@
+    recommended_action: 'Configure health check policies for the virtual gateway listeners in your App Mesh, specifying values for healthy threshold, health check interval, health check protocol, timeout period, and unhealthy threshold.',
+    apis: ['AppMesh:listMeshes', 'AppMesh:listVirtualGateways', 'AppMesh:describeVirtualGateway'],
+   
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var region = helpers.regions(settings);
+
+        async.each(region.appmesh, function(region, rcb) {
+            var listMeshes = helpers.addSource(cache, source,
+                ['appmesh', 'listMeshes', region]);
+
+            if (!listMeshes) return rcb();
+
+            if (listMeshes.err || !listMeshes.data) {
+                helpers.addResult(results, 3,
+                    'Unable to query for App Mesh meshes: ' + helpers.addError(listMeshes), region);
+                return rcb();
+            }
+
+            if (!listMeshes.data.length) {
+                helpers.addResult(results, 0, 'No App Mesh meshes found', region);
+                return rcb();
+            }
+
+            for (let mesh of listMeshes.data){
+                if (!mesh.arn || !mesh.meshName) continue;
+
+                let meshResource = mesh.arn;
+
+                var listVirtualGateways = helpers.addSource(cache, source,
+                    ['appmesh', 'listVirtualGateways', region, mesh.meshName]);
+
+                if (!listVirtualGateways || listVirtualGateways.err || !listVirtualGateways.data) {
+                    helpers.addResult(results, 3,
+                        'Unable to query for App Mesh virtual gateways: ' + helpers.addError(listVirtualGateways),
+                        region, meshResource);
+                    continue;
+                }
+
+                if (!listVirtualGateways.data.virtualGateways || !listVirtualGateways.data.virtualGateways.length) {
+                    helpers.addResult(results, 0,
+                        'No App Mesh virtual gateways found', region, meshResource);
+                    continue;
+                }
+
+                for (let gateway of listVirtualGateways.data.virtualGateways) {
+                    if (!gateway.arn || !gateway.virtualGatewayName) continue;
+
+                    let gatewayResource = gateway.arn;
+
+                    var describeVirtualGateway = helpers.addSource(cache, source,
+                        ['appmesh', 'describeVirtualGateway', region, gateway.virtualGatewayName]);
+
+                    if (!describeVirtualGateway ||
+                        describeVirtualGateway.err ||
+                        !describeVirtualGateway.data ||
+                        !describeVirtualGateway.data.virtualGateway ||
+                        !describeVirtualGateway.data.virtualGateway.spec ||
+                        !describeVirtualGateway.data.virtualGateway.spec.listeners) {
```

Comment:
We should have a separate check for length of listeners and if length is 0 then return a passing result